code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -131,18 +145,22 @@
 		activate_pin(3)
 		return
 
-	set_pin_data(IC_OUTPUT, 1, input_list[index])
+	var/key = input_list[index]
+	set_pin_data(IC_OUTPUT, 1, key)
+	if(istext(key))
+		set_pin_data(IC_OUTPUT, 2, input_list[key])
 	push_data()
 	activate_pin(2)
 
 
 /obj/item/integrated_circuit/lists/delete
 	name = "delete circuit"
-	desc = "This circuit will remove an element from a list by the index."
+	desc = "This circuit will remove an element from a list by the index if specified, and an element from the list with the matching key if specified for associative lists."
 	extended_desc = "If there is no element with such index, result list will be unchanged."
 	inputs = list(
 		"list" = IC_PINTYPE_LIST,
-		"index" = IC_PINTYPE_INDEX
+		"index" = IC_PINTYPE_INDEX,
```

Comment:
intentional this one only operates on numerical index in the list if you want to find something by the key use the search circuit

Reply:
It is kinda against entire idea of using keys,isn't it?
Why you even need lists with arguments this way?

Reply:
there's still uses of indexes instead of keys this lets them choose if you can do it why not there's legitimate uses for indexes like ordering.

Reply:
because entire point of keys is ability to use them for access to elements. Now you are trying to make just double lists,Without any advantages of associated lists.

Reply:
That's issue about another circuits too.

Reply:
so use the find circuit to use keys not this one they're meant to be separate index/keys.